control-plane/src/health.rs

```diff
@@ -88,42 +76,32 @@ async fn health_check_data_plane() -> HealthCheckLog {
 
 impl HealthCheckServer {
     pub async fn new() -> ServerResult<Self> {
-        let data_plane_checks_enabled = env_var_present_and_true!("DATA_PLANE_HEALTH_CHECKS");
         let tcp_server = TcpServer::bind(SocketAddr::from((
             [0, 0, 0, 0],
             CONTROL_PLANE_HEALTH_CHECK_PORT,
         )))
         .await?;
-        Ok(HealthCheckServer {
-            tcp_server,
-            config: HealthCheckServerConfig {
-                data_plane_checks_enabled,
-            },
-        })
+        Ok(HealthCheckServer { tcp_server })
     }
 
     pub async fn start(&mut self) -> ServerResult<()> {
         println!(
-            "Control plane health-check server running on port {CONTROL_PLANE_HEALTH_CHECK_PORT}. Also checking data-plane: {}", self.config.data_plane_checks_enabled
+            "Control plane health-check server running on port {CONTROL_PLANE_HEALTH_CHECK_PORT}"
         );
 
         loop {
             let stream = self.tcp_server.accept().await?;
-            let config = self.config.clone();
-            let service = hyper::service::service_fn(move |request: Request<Body>| {
-                let config = config.clone();
-                async move {
-                    match request
-                        .headers()
-                        .get("User-Agent")
-                        .map(|value| value.as_bytes())
-                    {
-                        Some(b"ECS-HealthCheck") => run_ecs_health_check_service(config).await,
-                        _ => Response::builder()
-                            .status(500)
-                            .body(Body::from("Unsupported health check type!"))
-                            .map_err(ServerError::from),
-                    }
+            let service = hyper::service::service_fn(move |request: Request<Body>| async move {
+                match request
+                    .headers()
+                    .get("User-Agent")
+                    .map(|value| value.as_bytes())
+                {
+                    Some(b"ECS-HealthCheck") => run_ecs_health_check_service().await,
+                    _ => Response::builder()
+                        .status(500)
```

Comment:
500 seems inappropriate here
```suggestion
                        .status(400)
```